clash-prelude/src/Clash/Explicit/BlockRam.hs

```diff
@@ -1111,19 +1114,25 @@ readNew clk rst en ram rdAddr wrM = mux wasSame wasWritten $ ram rdAddr wrM
                              (readNewT <$> rdAddr <*> wrM))
 
 
-data RamOp n a = RamRead (Index n) | RamWrite (Index n) a
+data RamOp n a = RamRead (Index n) | RamWrite (Index n) a | NoOp deriving(Generic,NFDataX)
 
 ramOpAddr :: RamOp n a -> Index n
 ramOpAddr (RamRead addr)    = addr
 ramOpAddr (RamWrite addr _) = addr
+ramOpAddr NoOp              = deepErrorX "Address for No operation undefined"
```

Comment:
Why `deepErrorX` instead of `errorX`? `Index n` isn't a (sum-of-)product type, so it's shallow.